zir/bits.zir

```diff
@@ -12,76 +12,8 @@ component NondetBitReg(val: Val) {
   reg
 }
 
-component BitReg(val: Val) {
-  reg := NondetBitReg(val);
-  val = reg;
-  reg
-}
-
-// Check that valid bits are valid
-test BitInRange {
-  AssertBit(0);
-  AssertBit(1);
-}
-
-// Check that 2 is not a bit
-test_fails BitOutOfRange {
-  AssertBit(2);
-}
-
-// Assert the a given value is a 2-bit number (i.e. 0-3)
-function AssertTwit(val: Val) {
-  val * (1 - val) * (2 - val) * (3 - val) = 0;
-}
-
 // Simple bit ops
+// Overrides the builtin version
```

Comment:
Why is this desirable?

Reply:
It was in the original bits.zir file. My guess is that it allows you to constrain the expression `x & y`. If you know that x and y are bits bitwise-and and multiplication are equivalent. So, as a "just in case" measure I left it there in case some of the poseidon code uses it